### ISSUE TITLE
Can't parser JSX

### ISSUE THREAD

**Issue by fisker** [meriyah/meriyah#176]
Self closing can parse

```js
> require('meriyah').parse('var  a = <div/>', {jsx: true}).body[0].declarations[0].init
{
  type: 'JSXElement',
  children: [],
  openingElement: {
    type: 'JSXOpeningElement',
    name: { type: 'JSXIdentifier', name: 'div' },
    attributes: [],
    selfClosing: true
  },
  closingElement: null
}
```

but

```js
require('meriyah').parse('var  a = <div></div>', {jsx: true}).body[0].declarations[0].init
Uncaught [ParseError [SyntaxError]: [1:20]: Unexpected token
] {
  index: 20,
  line: 1,
  column: 20,
  description: '[1:20]: Unexpected token',
  loc: { line: 1, column: 20 }
}
```

Did I do something wrong?

**Comment by 3cp** [comment]
It seems the latest 4.1.0 shipped with wrong dist files. I think this is the second time we got wrong dist files (maybe only the second time we noticed). I will make sure to add step to clean up tsc cache before bundle.

**Comment by 3cp** [comment]
It's probably from rollup-plugin-typescript2 https://github.com/ezolenko/rollup-plugin-typescript2/issues/262

I will add clean: true.

**Comment by 3cp** [comment]
I will release a patch tonight when I got back to home mac.

**Comment by 3cp** [comment]
v4.1.1 fixed the issue.

I think rollup-plugin-typescript2 has some bug on determine invalidating cache. One ts file may be affected by multiple ts files (especially those only provide typing, but yield no js file).

**Comment by fisker** [comment]
Thanks!

**Comment by agilgur5** [comment]
Hi 👋  . I help maintain `rollup-plugin-typescript2` nowadays and was looking into old issues and saw this was referenced from https://github.com/ezolenko/rollup-plugin-typescript2/issues/262 .

I haven't been able to reproduce this issue and don't think it's related to this problem. If you believe this has to do with type-only files per your previous comment

> I think rollup-plugin-typescript2 has some bug on invalidating cache. One ts file may be affected by multiple ts files (especially those only provide typing, but yield no js file).

then it could very well be due to https://github.com/ezolenko/rollup-plugin-typescript2/issues/7 instead, which indeed points out that the cache's module graph may not correctly set dependencies ("One ts file may be affected by multiple ts files") between modules when it comes to type-only files

Just thought I'd correct that for posterity and future reference, as well as give a possible root cause based on the comments here.

**Comment by agilgur5** [comment]
Wanted to update here that https://github.com/ezolenko/rollup-plugin-typescript2/issues/7 has been fixed by https://github.com/ezolenko/rollup-plugin-typescript2/pull/406 and released in rpt2 [`0.34.0`](https://github.com/ezolenko/rollup-plugin-typescript2/releases/tag/0.34.0) 🙂
